src/HDF5.jl

```diff
@@ -1094,6 +1112,28 @@ dataspace(sz::Dims{N}; max_dims::Union{Dims{N},Tuple{}}=()) where {N} = _dataspa
 dataspace(sz1::Int, sz2::Int, sz3::Int...; max_dims::Union{Dims,Tuple{}}=()) = _dataspace(tuple(sz1, sz2, sz3...), max_dims)
 
 
+function Base.ndims(dspace::Dataspace)
+    checkvalid(dspace)
+    return Int(h5s_get_simple_extent_ndims(dspace))
+end
+function Base.size(dspace::Dataspace)
+    checkvalid(dspace)
+    h5_dims = h5s_get_simple_extent_dims(dspace, nothing)
+    N = length(h5_dims)
+    return ntuple(i -> @inbounds(Int(h5_dims[N-i+1])), N)
+end
+function Base.size(dspace::Dataspace, d)
+    N = ndims(dspace)
+    d > N && return 1
+    h5_dims = h5s_get_simple_extent_dims(dspace, nothing)
+    return @inbounds Int(h5_dims[N - d + 1])
+end
+function Base.length(dspace::Dataspace)
+    checkvalid(dspace)
+    h5_dims = h5s_get_simple_extent_dims(dspace, nothing)
+    return Int(prod(h5_dims))
+end
+
 function get_dims(dspace::Dataspace)
```

Comment:
We should figure out a way to get rid of this and add it as a method of `size` or at least rename it

Reply:
Something like `size(obj; max = true)` or `size(obj, :max)` (for any spelling of `max`) to return both the size and max dims?

Reply:
Yeah that would be ideal. Wouldn't we run into the same issue with a type-unstable output  if we used `max=true`?

Reply:
I think so, yes.

Reply:
Yeah I don't think there's a good way to get rid of this by piggybacking on existing methods.  Unless we dispatch on `Val` types which is ugly and seems unnecessary. 


Why don't we rename it to `get_simple_extent_dims`  that way it adhere's  to the general api structure we are following with respect to naming.

Reply:
`get_dims` is also missing the `close` method on the dataspace.

Reply:
> Why don't we rename it to `get_simple_extent_dims` that way it adhere's to the general api structure we are following with respect to naming.

I'm OK with that; just unfortunate we didn't get there before tagging 0.14, but :shrug: I guess we could add a silent deprecation for a while.

> `get_dims` is also missing the close method on the `dataspace`.

I can edit that, too; also should be extended to `Union{Dataset,Attribute}` since everything else is defined over that union already.

Reply:
I realized, I was thinking of both `get_dims` methods — one on `Dataspace` and the other on `Dataset` — at the same time, but they don't have to be named the same. `get_dims(::Dataset)` is the method that's actually advertised in the docs. Were you thinking that only the `Dataspace` method should be renamed and we leave the `Dataset` method as-is?

Reply:
> I'm OK with that; just unfortunate we didn't get there before tagging 0.14, but shrug I guess we could add a silent deprecation for a while.


Yeah, I think it's fine though. I don't think we're 100% there yet with the external api yet, and I think it was a good idea to have an intermediate release(s) to get feedback and bug reports. Plus in the next minor release we can get rid of the depreciations added in this cycle.

Reply:
It's documented, but not exported and kind of 'internal exported api'. I think either (1) both should get renamed or (2) at least the method on  `Dataspace` and getting rid of the `Dataset` method `get_dims` method. I suspect most  users actually care about the `Base.size`  method, and determining `max_dims` is more for special use cases.